### PR TITLE
Added custom NumbersException class

### LINKED ISSUE #14: Custom exception class

### PR DESCRIPTION
**Pull request by jmulford-bandwidth**
closes #14

### CODE REVIEW

**Comment by jmulford-bandwidth** [comment]
I'll just make the NumbersSerde in this PR and close #17 as well to make the travis scripts happy

File: src/main/java/com/bandwidth/sdk/numbers/models/NumbersApiError.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableNumbersApiError.class)
+@JsonDeserialize(as = ImmutableNumbersApiError.class)
+public abstract class NumbersApiError {
```
Comment (fuchsnj) [inline src/main/java/com/bandwidth/sdk/numbers/models/NumbersApiError.java]: I think the Numbers error's look different than our Messages api. We should probably just remove this (And NumbersApiFieldError)

Comment (jmulford-bandwidth) [reply src/main/java/com/bandwidth/sdk/numbers/models/NumbersApiError.java]: I think that's right. Based on the docs, all of the responses look like this

```
<BillingReportCreationResponse>
    <ResponseStatus>
        <ErrorCode>15501</ErrorCode>
        <Description>Invalid date range. Start date 2014-05-21 must be before end date 2013-05-29</Description>
    </ResponseStatus>
</BillingReportCreationResponse>
```

All have the same ResponseStatus fields, but the outer tag looks different for all of them

File: src/main/java/com/bandwidth/sdk/numbers/exception/ExceptionUtils.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.bandwidth.sdk.numbers.exception;
+
+import java.util.concurrent.CompletableFuture;
+
+public class ExceptionUtils {
```
Comment (ccogan-band) [inline src/main/java/com/bandwidth/sdk/numbers/exception/ExceptionUtils.java]: I see this used pattern in Messaging SDK and Number SDK (with only the Exception name being different).  Might make sense to pull this out functionality to a shared lib.

**Comment by jmulford-bandwidth** [comment]
Already done

**Pull request closed without merging**